### PR TITLE
Update Magento_Braintree module ReadMe

### PR DESCRIPTION
**Pull request by diazwatson**
### Description (*)

Updates ReadMe for the Magento_Braintree module.

### Fixed Issues (if relevant)

1. magento/devdocs https://github.com/magento/devdocs/issues/4451: README content for modules - work in M2 Code

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by pmclain — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @pmclain, thank you for the review.
[ENGCOM-5920](https://jira.corp.magento.com/browse/ENGCOM-5920) has been created to process this Pull Request

**Comment by kalpmehta** [comment]
@diazwatson I don't see very detailed README.md file in any other module(s). If we add these details in one module's README, we may have to add it to other modules as well going forward to maintain consistency and that might become an overhead. Do you see this README template in any core module that you are following?

**Comment by diazwatson** [comment]
@kalpmehta have you seen the task description as well as all the PR (already merge) associated to this task? https://github.com/magento/devdocs/issues/4451?

**Comment by kalpmehta** [comment]
@diazwatson My bad, I missed the devdocs issue conversations. I will go ahead and approve this. I believe @jeff-matthews will have a final review on the content. Thanks for your contribution!

**Comment by diazwatson** [comment]
No worries, thanks :)

**Comment by engcom-Delta** [comment]
**Notice:** QA not applicable

File: app/code/Magento/Braintree/README.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+
+ - `payment_method_assign_data_braintree` event in `Magento\Braintree\Observer\DataAssignObserver` file.
+ - `payment_method_assign_data_braintree_paypal` event in `Magento\Braintree\Observer\DataAssignObserver` file.
+ - `shortcut_buttons_container` event in `Magento\Braintree\Observer\AddPaypalShortcuts` file.
+
+For information about an event in Magento 2, see [Events and observers](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/events-and-observers.html#events).
+
+### Layouts
+
+This module interacts the following layouts and layout handles in the `view/adminhtml/layout` directory:
```
Comment (jeff-matthews) [inline app/code/Magento/Braintree/README.md]:
```suggestion
This module interacts with the following layouts and layout handles in the `view/adminhtml/layout` directory:
```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]

File: app/code/Magento/Braintree/README.md (hunk 1 above)
Comment (sidolov) [reply app/code/Magento/Braintree/README.md]: Changes are applied

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-5920](https://jira.corp.magento.com/browse/ENGCOM-5920) has been created to process this Pull Request

**Review by jeff-matthews — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @jeff-matthews, thank you for the review.
[ENGCOM-5920](https://jira.corp.magento.com/browse/ENGCOM-5920) has been created to process this Pull Request

**Pull request merged**